specification/cognitiveservices/data-plane/Speech/SpeechToText/preview/v3.1-preview.1/speechtotext.json

```diff
@@ -0,0 +1,6237 @@
+          "201": {
+            "description": "The response contains information about the entity as payload and its location as header.",
+            "schema": {
+              "$ref": "#/definitions/Dataset"
+            },
+            "headers": {
+              "Location": {
+                "description": "The location of the created resource.",
+                "type": "string",
+                "format": "uri"
+              }
+            }
+          },
+          "default": {
+            "description": "An error occurred.",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        },
+        "deprecated": true,
+        "x-ms-examples": {
+          "Upload a dataset": {
+            "$ref": "./examples/upload_dataset.json"
+          }
+        }
+      }
+    },
+    "/datasets/uploadsessions": {
+      "post": {
+        "tags": [
+          "Custom Speech datasets for model adaptation:"
+        ],
+        "summary": "Create upload session.",
+        "operationId": "Datasets_CreateUploadSession",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "body",
+            "name": "dataset",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Dataset"
+            }
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "The response contains information about the entity as payload and its location as header.",
+            "schema": {
+              "$ref": "#/definitions/UploadSession"
+            },
+            "headers": {
+              "Location": {
+                "description": "The location of the created resource.",
+                "type": "string",
+                "format": "uri"
+              }
+            }
+          },
+          "default": {
+            "description": "An error occurred.",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create session to upload a dataset": {
+            "$ref": "./examples/create_upload_session.json"
+          }
+        }
+      }
+    },
+    "/datasets/uploadsessions/{id}": {
+      "put": {
+        "tags": [
+          "Custom Speech datasets for model adaptation:"
+        ],
+        "summary": "Upload a chunk of data.",
+        "operationId": "Datasets_UploadChunkIntoUploadSession",
+        "consumes": [
+          "application/octet-stream"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "id",
+            "required": true,
+            "type": "string",
+            "format": "uuid"
+          },
+          {
+            "in": "header",
+            "name": "Content-Range",
```

Comment:
Per [RFC7231](https://datatracker.ietf.org/doc/html/rfc7231#section-4.3.4), an origin server must respond with a 400 error for any request that contains a Content-Range header. However, I believe that we ma be violating that rule for other services.